src/services/pagination/Paginator.js

```diff
@@ -0,0 +1,196 @@
+import LesgoException from '../../exceptions/LesgoException';
+
+const FILE = 'Services/pagination/Paginator';
+
+export default class Paginator {
+  /**
+   * Constructor
+   *
+   * @param db
+   * @param sql
+   * @param sqlParams
+   * @param perPage
+   * @param currentPage
+   */
+  constructor(db, sql, sqlParams, perPage = 10, currentPage = 1) {
+    if (typeof perPage !== 'number') {
+      throw new LesgoException(
+        "Invalid type for 'perPage'",
+        `${FILE}::INVALID_TYPE_PER_PAGE`,
+        500,
+        { perPage }
+      );
+    }
+
+    if (typeof currentPage !== 'number') {
+      throw new LesgoException(
+        "Invalid type for 'currentPage'",
+        `${FILE}::INVALID_TYPE_CURRENT_PAGE`,
+        500,
+        { currentPage }
+      );
+    }
+
+    this.dbProp = db;
+    this.sqlProp = sql;
+    this.sqlParamsProp = sqlParams;
+    this.perPageProp = perPage;
+    this.currentPageProp = currentPage;
+
+    this.hasNext = false;
+
+    this.response = [];
+  }
+
+  /**
+   * Count all items in the current page.
+   *
+   * @returns {number}
+   */
+  async count() {
+    if (this.response.length <= 0) {
+      await this.executeQuery();
+    }
+
+    return this.response.length;
```

Comment:
return this.response.length || await this.executeQuery();

Reply:
Thanks @chocnut